pkg/endpoints/pipeline.go

```diff
@@ -166,6 +214,101 @@ func (r Resource) getPipelineRun(request *restful.Request, response *restful.Res
 	response.WriteEntity(pipelinerun)
 }
 
+func (r Resource) createPipelineRun(request *restful.Request, response *restful.Response) {
+	namespace := request.PathParameter("namespace")
+
+	pipelineRunData := ManualPipelineRun{}
+	if err := request.ReadEntity(&pipelineRunData); err != nil {
+		logging.Log.Error("error parsing request body on call to createPipelineRun %s", err)
+		utils.RespondError(response, err, http.StatusBadRequest)
+		return
+	}
+
+	createResponse := r.CreatePipelineRunImpl(pipelineRunData, namespace)
+	if createResponse.CODE == 204 {
+		response.WriteHeader(http.StatusNoContent)
+	} else { // anything other than 204 is an error - RespondError
+		utils.RespondError(response, createResponse.ERROR, createResponse.CODE)
+	}
+}
+
+/*CreatePipelineRunImpl - Create a new manual pipeline run and resources in a given namespace */
+func (r Resource) CreatePipelineRunImpl(pipelineRunData ManualPipelineRun, namespace string) *AppError {
+
+	pipelineName := pipelineRunData.PIPELINENAME
+	serviceAccount := pipelineRunData.SERVICEACCOUNT
+	regsitryLocation := pipelineRunData.REGISTRYLOCATION
+
+	startTime := getDateTimeAsString()
+
+	generatedPipelineRunName := fmt.Sprintf("tekton-pipeline-run-%s", startTime)
+
+	// Assumes you've already applied the yml: so the pipeline definition and its tasks must exist upfront.
+	pipeline, err := r.getPipelineImpl(pipelineName, namespace)
+	if err != nil {
+		errorMsg := fmt.Sprintf("could not find the pipeline template %s in namespace %s", pipelineName, namespace)
+		logging.Log.Errorf(errorMsg)
+		return &AppError{err, errorMsg, http.StatusPreconditionFailed}
+	}
+
+	logging.Log.Debugf("Found the pipeline template %s OK", pipelineName)
+
+	registryURL := regsitryLocation
```

Comment:
You're just changing the name of the existing variable here. Why not just name it `registryURL` to begin with? 😄